### ISSUE TITLE
crazydiskmark exits immediately on Ubuntu 20.04 with Wayland

### ISSUE THREAD

**Issue by ryanerwin** [fredcox/crazydiskmark#3]
Please let me know if I can provide any other information to help debug this.

```
 ❯ ./crazydiskmark
Starting...
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
Traceback (most recent call last):
  File "./crazydiskmark", line 10, in <module>
    window = crazydiskmark.MainWindow()
  File "/home/ryan/.local/lib/python3.8/site-packages/crazydiskmark/__init__.py", line 151, in __init__
    binaryDir = os.path.dirname(binaryFile)
  File "/usr/lib/python3.8/posixpath.py", line 152, in dirname
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

**Comment by fredcox** [comment]
Hi,

please, can u try this,

`sudo apt install libxcb-xinerama0`

and put ouput from crazydiskmark command here again?

**Comment by fredcox** [comment]
This is ubuntu with wayland. 

No problem here
![VirtualBox_Ubuntu 20_26_07_2020_13_46_41](https://user-images.githubusercontent.com/65084421/88484669-9f919500-cf46-11ea-87a4-a6ef97a7b8a9.png)

**Comment by fredcox** [comment]
You need upgrade the version too,

Please, do following command

`$ pip3 install crazydiskmark --upgrade`

To upgrade to the latest release

**Comment by ryanerwin** [comment]
> Hi,
> 
> please, can u try this,
> 
> `sudo apt install libxcb-xinerama0`
> 
> and put ouput from crazydiskmark command here again?

Interestingly libxcb-xinerama0 was already installed.

```
 ❯ sudo apt install libxcb-xinerama0
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libxcb-xinerama0 is already the newest version (1.14-2).
libxcb-xinerama0 set to manually installed.
0 upgraded, 0 newly installed, 0 to remove and 4 not upgraded.
```

When I try to launch crazydiskmark, it pretty instantly crashes...
```
 ❯ ~/.local/bin/crazydiskmark
Starting...
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
Traceback (most recent call last):
  File "/home/ryan/.local/bin/crazydiskmark", line 10, in <module>
    window = crazydiskmark.MainWindow()
  File "/home/ryan/.local/lib/python3.8/site-packages/crazydiskmark/__init__.py", line 154, in __init__
    binaryDir = os.path.dirname(binaryFile)
  File "/usr/lib/python3.8/posixpath.py", line 152, in dirname
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not NoneType

```

You can see that the `crazydiskmark` window starts to open for a moment, but then closes before it's done drawing...

![Peek 2020-07-27 06-41](https://user-images.githubusercontent.com/2156415/88492411-d9a97800-cfd4-11ea-9978-766fab3e8766.gif)

Here's the frame where the `crazydiskmark` window opens before immediately closing.

![image](https://user-images.githubusercontent.com/2156415/88492531-a1ef0000-cfd5-11ea-8d1f-03d1a3f52207.png)

**Comment by fredcox** [comment]
> razydiskmark.MainWindow()

The problem is **crazydiskmark.desktop** file is missing in your system. Please, create the **crazydiskmark.desktop** on **/home/ryan/.local/share/applications** directory with your favorite editor and put the following content:

[Desktop Entry]
Type=Application
Exec=/home/ryan/.local/bin/crazydiskmark
Path=/home/ryan/.local/bin
Comment=Crazy DiskMark is a utility to benchmark SSD disks on linux and produce results like CrystalDiskMark
Icon=/home/ryan/.local/lib/python3.8/site-packages/crazydiskmark/images/icon.png
Categories=System;
Name=Crazy DiskMark

save the file and start crazydiskmark again...

**Comment by fredcox** [comment]
> is **crazydiskmark.desktop** file is missing in your system. Please, create the **crazydiskmark.desktop**

I dont know why python is not creating automatically this file.

**Comment by ryanerwin** [comment]
Unfortunately that still doesn't seem to be it.

I'll try to find time to launch it in a brand new Ubuntu 20.04 VM and see what happens.

After creating `.desktop` file
```
 ❯ cat ~/.local/share/applications/crazydiskmark.desktop
[Desktop Entry]
Type=Application
Exec=/home/ryan/.local/bin/crazydiskmark
Path=/home/ryan/.local/bin
Comment=Crazy DiskMark is a utility to benchmark SSD disks on linux and produce results like CrystalDiskMark
Icon=/home/ryan/.local/lib/python3.8/site-packages/crazydiskmark/images/icon.png
Categories=System;
Name=Crazy DiskMark
```

Trying to launch `crazydiskmark` still exits almost immediately with the exact same window as in the screenshots posted earlier:

```
 ❯ ~/.local/bin/crazydiskmark
Starting...
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
Traceback (most recent call last):
  File "/home/ryan/.local/bin/crazydiskmark", line 10, in <module>
    window = crazydiskmark.MainWindow()
  File "/home/ryan/.local/lib/python3.8/site-packages/crazydiskmark/__init__.py", line 154, in __init__
    binaryDir = os.path.dirname(binaryFile)
  File "/usr/lib/python3.8/posixpath.py", line 152, in dirname
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

**Comment by fredcox** [comment]
Do you put **/home/ryan/.local/bin** in your **$PATH** environment variable?

This is my $PATH variable:

`/home/fredlins/.local/bin /usr/local/sbin /usr/local/bin /usr/sbin /usr/bin /sbin /bin /usr/games /usr/local/games /home/fredlins/nodejs/bin /snap/bin /home/fredlins/.local/bin /home/fredlins/android-studio/bin`

/home/ryan/.local/bin is must to be in your $PATH variable. 

Please put:

`export PATH=$PATH:/home/ryan/.local/bin`

in your **/home/ryan/.bashrc** file. And run crazydiskmark.

**Comment by ryanerwin** [comment]
I added `~/.local/bin` to my path, but still the same issue.

I tested it in a clean Fedora 32 VM that I just installed and CrazyDiskMark does work, so it's something on this system.
I'll try a few more tests later and see if I can isolate exactly what environmental difference is causing it and let you know.

![image](https://user-images.githubusercontent.com/2156415/88496920-d6b88280-cfe8-11ea-8755-4484b5d30ac3.png)
